test/gtest/ucs/test_async.cc

```diff
@@ -95,10 +94,27 @@ class base_event : public base_async {
         ucs_async_pipe_push(&m_event_pipe);
     }
 
+    void push_event_if_valid() {
+        if (m_event_valid == 1) {
+            push_event();
+        }
+    }
+
     void reset() {
         ucs_async_pipe_drain(&m_event_pipe);
     }
 
+    void create_event() {
+        ucs_status_t status = ucs_async_pipe_create(&m_event_pipe);
```

Comment:
Q: What happens when we call this function on already "created" event?
Shouldn't we check if object is already initialized, and if so - release the previous state first?

Reply:
IMO, double initialization is an error.
The life time of event is managed by `base_event`, since we call create/destroy method inside the constructor/destructor.
``` c
    base_event(ucs_async_mode_t mode) : base_async(mode) {
        create_event();
    }
    ...
    virtual ~base_event() {
        destroy_event();
    }
```
To avoid double initialization, I added assertion inside the `create_event` method, see
``` c
EXPECT_EQ(0, ucs_atomic_cswap32(&m_event_valid, 0, 1));
```
In addition, if we explicitly call `create_event`, which means we want to manage the event by ourself, then we're responsible for `destroy_event`.